### ISSUE TITLE
´msstore publish´ hangs on Submission Status - CommitStarted

### ISSUE THREAD

**Issue by meakbiyik** [microsoft/msstore-cli#29]
Both in my workstation and CI, the `msstore publish -i ".\release\build"` command hangs on "Submission Status - CommitStarted". When I look at the partner center, I see that although the previous (manual) submission works without issues, the new submission by CLI appears buggy, see "Pricing and Availability" warning below. Waiting over 1 hour does not change the status. There are no other errors.

![image](https://github.com/microsoft/msstore-cli/assets/23424198/29b89a90-3e94-4b12-9541-ab3eab024d27)

**Comment by meakbiyik** [comment]
There does not seem to be anything out of ordinary in the logs when I run the command with ´-v´, only the following warnings repeat:

![image](https://github.com/microsoft/msstore-cli/assets/23424198/fcadcdd4-4653-48aa-97fd-0930cbe8d389)

**Comment by azchohfi** [comment]
This is the normal execution of the publishing process. It will stay at Status=CommitStarted until it passes the first steps of certification, which might take a while. The CLI pools every 30 seconds, until it changes to something else, then it report it.

**Comment by meakbiyik** [comment]
Oh that I am aware, but should I expect it to take over 1 hour? 

<img width="979" alt="image" src="https://github.com/microsoft/msstore-cli/assets/23424198/e8097b37-58c1-4256-bcf8-5f7663d8d5fd">

The behavior was consistent in my local and the Github Action, which is sadly not acceptable if we want to automate the publishing process.

**Comment by meakbiyik** [comment]
> This is the normal execution of the publishing process. It will stay at Status=CommitStarted until it passes the first steps of certification, which might take a while. The CLI pools every 30 seconds, until it changes to something else, then it report it.

I also found one point in your comment interesting: does the `CommitStarted` status wait until the first steps of the certification process is concluded? Because the issue is I report appear to happen before the certification starts, since the submission does not appear in a `submitted for certification` state on the user interface, see that the `submit to the store` button in the first figure I added still appears (yet disabled). Below is the UI for a properly submitted package:

<img width="1032" alt="image" src="https://github.com/microsoft/msstore-cli/assets/23424198/be9d02a0-c99d-4bfb-87f2-ab2d11b04805">

My guess is that CLI configures the package metadata (most probably some field in the `pricing and availability` section) erroneously and therefore the package cannot be submitted to certification, and the app gets stuck in the `commit started` phase.

**Comment by azchohfi** [comment]
Are you still facing this issue?

**Comment by meakbiyik** [comment]
Yup, so I needed to disable auto-publish in the pipeline. Did you push a fix for it? I can test in the next release.

**Comment by azchohfi** [comment]
This seems to have been an intermittent issue with the service. I would try again, since we published a number of new version.

**Comment by bitbound** [comment]
I have the same issue.  It never progresses passed "Commit Started".  I have to go into the Partner Center and manually submit it.

After that, the command (and my DevOps pipeline) completes successfully.

I'm not sure if this is significant, but when I went to the Partner Center to check the submission, my new package was in the submission, but it still said "Unchanged" next to packages.  I went in there to double-check that it was the correct version, clicked Save, and then it showed as "Updated"

![image](https://github.com/microsoft/msstore-cli/assets/20995508/1626fb0b-1b86-4fcc-bfee-75e7bcfd0fc3)

**Comment by brandonp2412** [comment]
Yep mines stuck in a loop waiting for "Submit for certification" to be clicked in the Microsoft Partners website. After I did that manually the tool said it was successful and stopped.

**Comment by itm-github** [comment]
Are there any updates for this issue? Are anyone found a solution? I got the same problem an about two months. I create the app in the portal and published it. Next I was try to automate the building and publishing app with pipeline but the same issue as in this thread. When I publish the Submission - it stuck on CommitStarted status and after a day or two finished with CommitFailed status. It is interesting that if I just create a new submission (with Powershell StoreBroker module) next publish the Submission without any change - the Submission successfully goes into PreProcessing and next states. If I change even one field in the Submission (targetPublishMode, for example) - the Submission stuck on CommitStarted state. This same behaviour I got when using Powershell StoreBroker module and direct requests to the API with Python.

**Comment by bitbound** [comment]
> Are there any updates for this issue? Are anyone found a solution? I got the same problem an about two months. I create the app in the portal and published it. Next I was try to automate the building and publishing app with pipeline but the same issue as in this thread. When I publish the Submission - it stuck on CommitStarted status and after a day or two finished with CommitFailed status. It is interesting that if I just create a new submission (with Powershell StoreBroker module) next publish the Submission without any change - the Submission successfully goes into PreProcessing and next states. If I change even one field in the Submission (targetPublishMode, for example) - the Submission stuck on CommitStarted state. This same behaviour I got when using Powershell StoreBroker module and direct requests to the API with Python.

I haven't.  I've been going into Partner Center and manually submitting every time.

**Comment by CL-vc** [comment]
After experiencing this issue for a while, it seems to have been resolved somehow. While the exact cause is unclear, the process of uploading a .msix file to the Microsoft Store through the CLI now successfully finishes, and submits the submission for certification, at least for me.

**Comment by maxded** [comment]
I've recently started experiencing the same issue. For the past six months, my CI pipeline has consistently packaged my project as an MSIX and submitted it to the Microsoft Store using the `msstore publish` command. However, a few weeks ago, this process broke: the command now gets stuck at `Submission Status - CommitStarted`, as others have reported. This means that a draft submission is created but no longer proceeds automatically to certification as it did before. Previously, `msstore publish` would complete in about 10 minutes. Now, it runs for hours without progressing past the `CommitStarted` status—strongly suggesting something is wrong.

Another concerning issue is that when the submission gets stuck at `CommitStarted`, the resulting draft submission unexpectedly resets the base price to **Free**. This is especially dangerous because the draft's overview misleadingly claims that the "Pricing and availability" settings are unchanged, even though they clearly are not.

![Image](https://github.com/user-attachments/assets/15da6906-800c-4267-ab9a-bc799524b1d6)

![Image](https://github.com/user-attachments/assets/8e930f9e-5c6c-4eaa-b31d-e69e934df90a)

I'm not sure if these issues are directly related or if they're problems with the msstore CLI tool itself, but I wanted to mention them here since they seem similar to what others have reported.

**Comment by Diegorro98** [comment]
Same here, with StoreBroker, the whole publishing process only took about ~2'5 minutes, and with msstore cli, it takes from 5 minutes to one hour (because of token expired) and sometimes it does take even less than a minute (surprisingly).

**Comment by Zofware** [comment]
I'm also hitting an infinite loop on `Submission Status - CommitStarted`. On every iteration, I see a `SalesUnsupportedWarning`. That might be red herring. Regardless, my submissions get stuck.

For example,
```
...
2026-05-01T19:46:46.1746949Z Uploading Bundle to Azure blob 2%
2026-05-01T19:46:47.3262937Z ✅ Successfully uploaded the application package.
2026-05-01T19:46:47.3282617Z 07:46:47 info System.Net.Http.HttpClient.Default.LogicalHandler[100] => HTTP 
2026-05-01T19:46:47.3283242Z POST 
2026-05-01T19:46:47.3283891Z https://manage.devcenter.microsoft.com/v1.0/my/applications/appId-redacted/flights
2026-05-01T19:46:47.3284571Z /flightId-redacted/submissions/submissionId-redacted/commit 
2026-05-01T19:46:47.3285057Z Start processing HTTP request POST 
2026-05-01T19:46:47.3285526Z https://manage.devcenter.microsoft.com/v1.0/my/applications/appId-redacted/flights
2026-05-01T19:46:47.3286444Z /flightId-redacted/submissions/submissionId-redacted/commit
2026-05-01T19:46:47.3287298Z 07:46:47 info System.Net.Http.HttpClient.Default.ClientHandler[100] => HTTP 
2026-05-01T19:46:47.3287898Z POST 
2026-05-01T19:46:47.3290233Z https://manage.devcenter.microsoft.com/v1.0/my/applications/appId-redacted/flights
2026-05-01T19:46:47.3291100Z /flightId-redacted/submissions/submissionId-redacted/commit 
2026-05-01T19:46:47.3291639Z Sending HTTP request POST 
2026-05-01T19:46:47.3292156Z https://manage.devcenter.microsoft.com/v1.0/my/applications/appId-redacted/flights
2026-05-01T19:46:47.3292726Z /flightId-redacted/submissions/submissionId-redacted/commit
2026-05-01T19:47:10.6296158Z 07:47:10 info System.Net.Http.HttpClient.Default.ClientHandler[101] => HTTP 
2026-05-01T19:47:10.6297160Z POST 
2026-05-01T19:47:10.6297636Z https://manage.devcenter.microsoft.com/v1.0/my/applications/appId-redacted/flights
2026-05-01T19:47:10.6298224Z /flightId-redacted/submissions/submissionId-redacted/commit 
2026-05-01T19:47:10.6298692Z Received HTTP response headers after 23300.7613ms - 202
2026-05-01T19:47:10.6299408Z 07:47:10 info System.Net.Http.HttpClient.Default.LogicalHandler[101] => HTTP 
2026-05-01T19:47:10.6299959Z POST 
2026-05-01T19:47:10.6300416Z https://manage.devcenter.microsoft.com/v1.0/my/applications/appId-redacted/flights
2026-05-01T19:47:10.6300928Z /flightId-redacted/submissions/submissionId-redacted/commit End
2026-05-01T19:47:10.6301403Z processing HTTP request after 23301.5636ms - 202
2026-05-01T19:47:10.6310463Z Waiting for the submission commit processing to complete. This may take a couple
2026-05-01T19:47:10.6312029Z of minutes.
2026-05-01T19:47:10.6312691Z Submission Committed - Status=CommitStarted
2026-05-01T19:47:40.6414672Z 07:47:40 info System.Net.Http.HttpClient.Default.LogicalHandler[100] => HTTP 
2026-05-01T19:47:40.6415408Z GET 
2026-05-01T19:47:40.6416066Z https://manage.devcenter.microsoft.com/v1.0/my/applications/appId-redacted/flights
2026-05-01T19:47:40.6417219Z /flightId-redacted/submissions/submissionId-redacted/status 
2026-05-01T19:47:40.6417810Z Start processing HTTP request GET 
2026-05-01T19:47:40.6418376Z https://manage.devcenter.microsoft.com/v1.0/my/applications/appId-redacted/flights
2026-05-01T19:47:40.6419346Z /flightId-redacted/submissions/submissionId-redacted/status
2026-05-01T19:47:40.6420214Z 07:47:40 info System.Net.Http.HttpClient.Default.ClientHandler[100] => HTTP GET
2026-05-01T19:47:40.6420889Z https://manage.devcenter.microsoft.com/v1.0/my/applications/appId-redacted/flights
2026-05-01T19:47:40.6421537Z /flightId-redacted/submissions/submissionId-redacted/status 
2026-05-01T19:47:40.6422102Z Sending HTTP request GET 
2026-05-01T19:47:40.6422619Z https://manage.devcenter.microsoft.com/v1.0/my/applications/appId-redacted/flights
2026-05-01T19:47:40.6423189Z /flightId-redacted/submissions/submissionId-redacted/status
2026-05-01T19:47:54.9494766Z 07:47:54 info System.Net.Http.HttpClient.Default.ClientHandler[101] => HTTP GET
2026-05-01T19:47:54.9495950Z https://manage.devcenter.microsoft.com/v1.0/my/applications/appId-redacted/flights
2026-05-01T19:47:54.9496713Z /flightId-redacted/submissions/submissionId-redacted/status 
2026-05-01T19:47:54.9497228Z Received HTTP response headers after 14307.5418ms - 200
2026-05-01T19:47:54.9498012Z 07:47:54 info System.Net.Http.HttpClient.Default.LogicalHandler[101] => HTTP 
2026-05-01T19:47:54.9498541Z GET 
2026-05-01T19:47:54.9499044Z https://manage.devcenter.microsoft.com/v1.0/my/applications/appId-redacted/flights
2026-05-01T19:47:54.9499587Z /flightId-redacted/submissions/submissionId-redacteds/status End
2026-05-01T19:47:54.9500074Z processing HTTP request after 14308.378ms - 200
2026-05-01T19:47:54.9509170Z Submission Status - CommitStarted
2026-05-01T19:47:54.9643156Z 07:47:54 info: MSStore.CLI.ProjectConfigurators.MSIXProjectPublisher[0] 
2026-05-01T19:47:54.9643942Z SalesUnsupportedWarning - The sales resource is no longer supported. To view or 
2026-05-01T19:47:54.9644651Z edit the sales data for this submission, use the Dev Center dashboard.
...
```
(repeats infinitely)

**Comment by Zofware** [comment]
Having said the above, I did just have a submission succeed. Sample size of one disclaimer aside, the difference seems to be whether there was already a pending update in the Store or not. The submission that failed occurred when there was a pending update. After the failure, I deleted the pending update and ran the submission again. Then it succeeded. 🤷
